procrastinate/tasks.py

```diff
@@ -89,6 +92,10 @@ def __init__(
         self.name: str = name if name else self.full_path
         self.pass_context = pass_context
 
+        if aliases is not None:
+            assert isinstance(aliases, list)
```

Comment:
Hm ?
- Why do you request an "Iterable" and not a list, if you want to make sure it's a list ?
- Why not accept any iterable and just make it a list ?

Reply:
The first point is a mistake, initially I wanted to accept any iterable

Reply:
For the second point, if we do that, then we should ensure the argument is not a string, right ?

Reply:
Yes, if we consider we can accept "strings or lists" then I'd say we should do:
```python
if isinstance(value, str):
    value = [value]   # transform a single string into a list of strings

value = list(value)  # transform any iterable into a list
```

Reply:
This approch is fine to me, but I've to tell that my initial choice of restricting only to list was driven by the fear someone could pass an iterable object which was not a list, tuple, or str, and it could lead to an unexpected behaviour. That's why I was prefering a defensive check. But maybe it's not really relevant, what is your opinion about this ?

Reply:
In pythonic code, I'd say it's expected that we could receive any iterable. If we need a list internally, it's our job to make a list from the iterable.

That being said, it could also be ok to explicitely request a list, but in this case, I would document it (through type hints, which will find their way to the documentation) and... I wouldn't necessarily check the type at runtime: we'll be using duck typing, if someone give us something that "walks and quacks like a list", then for all intended purposes it's a list, we have no reason to tell them it's wrong.
We're in a dynamic language, we could start doubting every argument type of every function. If we document that we receive a single type, for me we can trust the user to send the type.

But the best it to request an Iterable, and then make it a list. That way, no questions asked.

Reply:
I'll keep the `List` approach as it's not blocking for you. Thanks for your input @ewjoachim.